_posts/a-larasati/2020-09-22-larasati-input-treasure-hunt.md

```diff
@@ -0,0 +1,104 @@
+
+The first thing I did at this point was decide how to approach the "within 5 pixels" part because in the code I wrote, "5 pixels" was the deciding factor. But what exactly did "5 pixels" mean? Does it mean that if the treasure was hidden on (10,10) and the user enters (5,5) or (15,15) or any combination in between, they win? Or does it mean if the treasure was on (10,10) they can enter any number for X and Y coordinate so long as the sum of the square of the values equals to 25 i.e. the square of 5 (if that makes any sense? The pythagorean theorem??)?
+
+I decided that "5 pixels" meant 5 steps: if the treasure is on (10,10) and the user enters (10,15) or (15,10) or (12,13) or (8,7) or any number combination that is a total of 5 steps away from the winning coordinates, the user wins the game. To do this, I decided to sum the absolute value of the `randint()` that was assigned to the run of the program and compare it to the sum of the absolute value of coordinate numbers the user enters. If the difference of these absolute values is less than or equal to 5, the user wins. I also normalized the value to reduce the chances of error in calculation by adding 100; the range of "correct" inputs taken is from -100 to 100, which means there are 201 possible numerical inputs (including 0), and adding 100 simply pushes the range from 0 to 200. Note: `treasure_x` and `treasure_y` are variables assigned to the `randint()` values from the base code, which represents the X and Y coordinates of the treasure.
+
+Regarding the feedback, I wanted the feedback to be relatively straightforward: using a standard X,Y coordinate plane and a `if` `else` statement, if the X value the user inputs is too high or too low, the feedback tells them that they are too far to the right or left, respectively. For Y values too high or too low, the feedback tells them that they are too high or low, respectively. If they found one correct coordinate but not the other, the feedback tells them they found that correct coordinate and their position relative to the incorrect one.
+
+```python
+#Normalizing the coordinates to prep it for the checks
+norm_xcoord = xcoord + 100
+norm_ycoord = ycoord + 100
+norm_treasurex = treasure_x + 100
+norm_treasurey = treasure_y + 100
+
+#Checks the input coordinates and compares it to the random X/Y generated
+check_xcoord = abs(norm_xcoord - norm_treasurex)
+check_ycoord = abs(norm_ycoord - norm_treasurey)
+
+if check_xcoord + check_ycoord > 5:
+	if xcoord > treasure_x:
+		print("You're too far to the right.")
+	if xcoord < treasure_x:
+		print("You're too far to the left.")
```

Comment:
I think the wording of the feedback can be in a way to show the user if they are doing the right move or not. For example, "you are getting close" or "you are going far" can provide the user with a more awareness about the consequence of their action